packages/core/test/util/renderer.test.ts

```diff
@@ -588,3 +589,100 @@ test('mapStateToLayoutProps - hidden via state with path from ownProps ', t => {
   const props = mapStateToLayoutProps(state, ownProps);
   t.false(props.visible);
 });
+
+test('should assign defaults to enum', t => {
+  const schema: JsonSchema = {
+    type: 'object',
+    properties: {
+      name: {
+        type: 'string',
+        minLength: 1
+      },
+      color: {
+        type: 'string',
+        enum: ['red', 'green', 'blue'],
+        default: 'green'
+      }
+    }
+  };
+
+  const uischema: UISchemaElement = undefined;
+
+  const data = {
+    name: 'foo'
+  };
+
+  const initState = {
+    jsonforms: {
+      core: {
+        uischema,
+        schema,
+        data,
+        errors: [] as ErrorObject[]
+      }
+    }
+  };
+  const store: Store<JsonFormsState> = createStore(
+    combineReducers({ jsonforms: jsonformsReducer() }),
+    initState
+  );
+  store.dispatch(
+    init(data, schema, uischema, createAjv({ useDefaults: true }))
+  );
+  const ownProps: OwnPropsOfControl = { schema };
+
+  mapDispatchToArrayControlProps(store.dispatch, ownProps);
+
+  t.is(store.getState().jsonforms.core.data.color, 'green');
+});
+
+test('should assign defaults to array', t => {
```

Comment:
A better name would be `"should assign defaults within nested object of an array"`

Reply:
Done, changed the name.